ipatests/pytest_ipa/integration/__init__.py

```diff
@@ -478,3 +480,18 @@ def del_compat_attrs(cls):
         del cls.ad_subdomains
         del cls.ad_treedomains
     del cls.ad_domains
+
+
+def skip_if_fips(reason='Not supported in FIPS mode', host='master'):
```

Comment:
@wladich 
Why we are not using pyest skipif decorator (https://docs.pytest.org/en/reorganize-docs/new-docs/user/skipping.html) with call for fips.is_fips_enabled? 
May be there could be scenario where we need to skip tests in FIPS mode due to some bug, so in that case it won't fit.

Reply:
@kaleemsiddiqu 

* `skipif` requires a boolean (not a callable) as as condition. 
* `is_fips_enabled` requires `host` argument, an instance of Host
*  the decorator is executed when the test class (not the instance) is created. 
*  Host objects are created in `mh` fixture, which is called for the test class instance 

=> When decorator is executed the Host object for test machine does not yet exist. 


I do not fully get the second part of the comment. We actually have a bug here and we skip the test until it is fixed.

Reply:
Second part is about specific reason displayed e.g skipping due to a bz which i think can be used in the current implementation.